lib/metasploit/framework/data_service/proxy/workspace_data_proxy.rb

```diff
@@ -20,47 +21,67 @@ def add_workspace(workspace_name)
 
   def default_workspace
     begin
-      data_service = self.get_data_service()
-      data_service.default_workspace
+      ws = find_workspace(Msf::DBManager::Workspace::DEFAULT_WORKSPACE_NAME)
+      if ws.nil?
+        ws = add_workspace(Msf::DBManager::Workspace::DEFAULT_WORKSPACE_NAME)
+      end
+      ws
     rescue  Exception => e
       self.log_error(e, "Problem finding default workspace")
     end
   end
 
   def workspace
     begin
-      data_service = self.get_data_service()
-      data_service.workspace
+      if @current_workspace
+        @current_workspace
+      else
+        # This is mostly a failsafe to prevent bad things from happening. @current_workspace should always be set
+        # outside of here, but this will save us from crashes/infinite loops if that happens
+        warn "@current_workspace was not set. Setting to default_workspace"
```

Comment:
Minor: should we ever end up in this code, would it be a good idea to output the name of the `default_workspace`? "@current_workspace was not set; assigning default_workspace #{default_workspace.name}"